catalog/core/catalog-core-api/src/main/java/ddf/catalog/plugin/PrePolicyPlugin.java

```diff
@@ -0,0 +1,94 @@
+package ddf.catalog.plugin;
+
+import java.util.Map;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.UpdateRequest;
+
+/**
+ * The {@link PrePolicyPlugin} extension point allows for request/response processing before any
+ * policies are implemented at the very beginning of the plugin chain.
+ * <p>
+ * FOR INTERNAL USE ONLY
```

Comment:
This plugin is for public use so this should not be marked like this.  But you should leave the experimental warning.